File: src/experiments/monolith/javascript/wmh/ex/Rectangle.mjs
import {Shape} from './Shape.mjs';

export class Rectangle extends Shape {

  width() { return this._width; }
  height() { return this._height; }

  constructor(width, height) {
    super();
    this._width = width;
    this._height = height;
  }

  area() {
    return this.width() * this.height();
  }
}

File: src/experiments/monolith/cpp/wmh/ex/Rectangle.cc
#include "Rectangle.hh"

namespace wmh {
namespace ex {

Rectangle::Rectangle(float width, float height) {
  this->_width = width;
  this->_height = height;
}

float
Rectangle::area() const {
  return this->_width * this->_height;
}

}  // namespace ex
}  // namespace wmh
